lib/checker.js

```diff
@@ -350,124 +374,73 @@ function prepareImagesForComparison(twoDimensionalArrayOfBuffers) {
-			originalImage: {
-				buffer: bufferPreppedOriginal
-			},
-			reproducedImage: {
-				buffer: bufferPreppedReproduction
-			}
-		};
-		this.resizeResultCode = resultCode;
-	}
+function resizeImageIfNecessary(originalImageBuffer, reproducedImageBuffer, dimensionsOriginal, dimensionsReproduced) {
 
 	return new Promise(
 		function (resolve, reject) {
 			if (dimensionsOriginal.width != dimensionsReproduced.width || dimensionsOriginal.height != dimensionsReproduced.height) {
-				if (dimensionsOriginal.width != 1344 || dimensionsOriginal.height != 960) {
-					sharp(originalImage)
-						.resize(1344, 960)
-						.png()
-						.toBuffer()
-						.then(originalResized => {
-							debugCompare("Resizing Original image No.%s.".yellow, index);
-							originalImage = originalResized;
-							if (dimensionsReproduced.width != 1344 || dimensionsReproduced.height != 960) {
-								sharp(reproducedImage)
-									.resize(1344, 960)
-									.png()
-									.toBuffer()
-									.then(reproResized => {
-										debugCompare("Resized Reproduced image No.%s.".yellow, index);
-										reproducedImage = reproResized;
-										resultHandler(true);
-									})
-									.catch(e => {
-										debugERROR("Failure resizing Reproduced image No.%s.".red, index);
-										metadata.errors.push(e);
-										resultHandler(false, e);
-									});
-							}
-						})
-						.catch(e => {
-							debugERROR("Failure resizing Original image No.%s.".red, index);
-							metadata.errors.push(e);
-							resultHandler(false, e)
-						});
-				}
-				else {
-					if (dimensionsReproduced.width != 1344 || dimensionsReproduced.height != 960) {
-						sharp(reproducedImage)
-							.resize(1344, 960)
-							.png()
-							.toBuffer()
-							.then(reproducedResized => {
-								debugCompare("Resized Reproduced image No.%s.".yellow, index);
-								reproducedImage = reproducedResized;
-								resultHandler(true);
-							})
-							.catch(e => {
-								debugERROR("Failure resizing Reproduced image No.%s.".red, index);
-								metadata.errors.push(e);
-								resultHandler(false, e);
-							});
-					}
-				}
-			}
-			else {
-				debugCompare("No resizing needed for images with index %s", index);
-				resolve(new ResizeOperationResults(originalImage, reproducedImage, 0));
+				resizing(reproducedImageBuffer, dimensionsOriginal, (reproducedImageBufferResized) => {
+					resolve(templateBuffer(originalImageBuffer, reproducedImageBufferResized));
+				});
 			}
 
-			function resultHandler(resizingSuccessfull) {
-
-				if (resizingSuccessfull) {
-					resolve(new ResizeOperationResults(originalImage, reproducedImage, 1));
-				}
-				else {
-					resolve(new ResizeOperationResults(originalImage, reproducedImage, 2));
-				}
+			else {
+				debugCompare("No resizing needed for images");
+				resolve(templateBuffer(originalImageBuffer, reproducedImageBuffer));
 			}
 		}
 	)
 }
 
 /**
- *
+ * Function which compares each of the image pairs by pixelmatch
```

Comment:
"compares .. with pixelmatch" or "by using" ?

Reply:
Wrote a new comment which is more clear